### PR TITLE
SLE-340 Bump sonarlint-core to 4.8

### PR DESCRIPTION
**Pull request by jblievremont**

### CODE REVIEW

File: org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/analysis/IAnalysisConfigurator.java [hunk 1]
```diff
@@ -51,12 +52,24 @@ default void analysisComplete(IPostAnalysisContext context, IProgressMonitor mon
...
   /**
    * By default SonarLint Eclipse will exclude some analyzers that are not working out of the box without some requirements.
-   * If this IAnalysisConfigurator is fullfilling some requirements, then you can whitelist the associated plugin.
+   * If this IAnalysisConfigurator is fulfilling some requirements, then you can whitelist the associated plugin.
    * @return List of plugin keys that should be whitelisted (e.g. java, cpp, cobol, ...)
    * @since 4.2
+   * @deprecated starting with 4.4, IAnalysisConfigurator should provide explicitly enabled languages instead
+   * @see #whitelistedLanguages()
    */
   default Set<String> whitelistedPlugins() {
```
Comment (henryju) [inline org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/analysis/IAnalysisConfigurator.java]: Missing `@Deprecated` annotation

File: org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/server/Server.java [hunk 2]
```diff
@@ -106,18 +106,19 @@ public static String getSonarCloudUrl() {
   Server(String id) {
     this.id = id;
 
-    Set<String> excludedPlugins = new HashSet<>(Arrays.asList("typescript", "java", "cpp"));
+    EnumSet<Language> languagesDisabledByDefault = EnumSet.of(Language.TS, Language.JAVA, Language.CPP);
```
Comment (henryju) [inline org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/server/Server.java]: I think at some point we should plan to disable Cobol by default. But for that we would have to require our partner to whitelist Cobol on their side. Maybe we could create a ticket to not forget.

Comment (henryju) [inline org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/server/Server.java]: C and OBJECTIVEC should be disabled by default

Comment (henryju) [reply org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/server/Server.java]: Swift should also be disabled (see the matrix)

File: org.sonarlint.eclipse.cdt/src/org/sonarlint/eclipse/cdt/internal/CProjectConfiguratorExtension.java [hunk 3]
```diff
@@ -62,6 +63,14 @@ private static boolean isCdtPresent() {
     return Collections.emptySet();
   }
 
+  @Override
+  public Set<Language> whitelistedLanguages() {
+    if (isCdtPresent()) {
+      return Collections.singleton(Language.CPP);
```
Comment (henryju) [inline org.sonarlint.eclipse.cdt/src/org/sonarlint/eclipse/cdt/internal/CProjectConfiguratorExtension.java]: Also whitelist C

File: org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/server/Server.java (hunk 2 above)
Comment (jblievremont) [reply org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/server/Server.java]: https://jira.sonarsource.com/browse/SLE-341

**Comment by jblievremont** [comment]
Manually merged.

**Pull request closed without merging**